hypershift-operator/controllers/nodepool/haproxy.go

```diff
@@ -403,6 +414,63 @@ func generateHAProxyStaticPod(name, image, internalAPIAddress, configPath string
 	}
 }
 
+func generateNodePublicInfoViewerCR() func() ([]byte, error) {
```

Comment:
I don't know if this is the correct location to do this.  This RBAC is cluster scope and not nodepool scoped.  I suspect it belongs in the HCCO along side reconciling the in cluster Kubernetes API server endpoint.  Please ask Red Hat for their opinion.